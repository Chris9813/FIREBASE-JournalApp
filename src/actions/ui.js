import { types } from "../types/types";

export const setError = (err) => ({
    type: types.uiSetError,
    payload: err,
})

export const remError = (err) => ({
    type: types.uiRemoveError,
})

