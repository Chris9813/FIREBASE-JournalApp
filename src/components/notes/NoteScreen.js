import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote, startDeleting } from '../../actions/notes'
import { useForm } from '../hooks/useForm'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    const dispatch = useDispatch();

    const {active:note} = useSelector(state => state.notes)
    const [values, handleInputChange, reset] = useForm(note)
    
    const {body, title, id} = values

    const activeId = useRef( note.id )

    useEffect(() => {
        if(note.id !== activeId.current){
            reset(note)
            activeId.current = note.id
        }
    }, [note, reset])

    useEffect(() => {
        dispatch(activeNote(values.id, {...values}))
        
    }, [values, dispatch])

    const handleDelete = () => {
        dispatch(startDeleting(id));
    }

    return (
        <div className="notes_main-content">
            <NotesAppBar />

            <div className="notes_content">
                <input 
                type="text"
                placeholder="Another awsome title"
                className="notes_title-input"
                autoComplete="off"
                value = {title}
                onChange = {handleInputChange}
                name="title"
                />

                <textarea
                placeholder="What happened today"
                className="notes_textarea"
                value={body}
                onChange={handleInputChange}
                name="body"
                >
                </textarea>

                {
                note.url &&
                <div className="notes_image">
                    <img 
                    src={note.url}
                    alt="Imagen"
                    height= "150px"
                    />
                </div>
                }
            </div>
            <button className="btn btn-danger"
            onClick={handleDelete}
            >
                    Delete
            </button>
        </div>
    )
}
