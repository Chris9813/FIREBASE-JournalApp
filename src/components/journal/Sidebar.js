import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startLogout } from '../../actions/auth';
import { noteLogout, startNewNote } from '../../actions/notes';
import { JournalEntries } from './JournalEntries';


export const Sidebar = () => {

    const dispatch = useDispatch();

    const handleLogout = () => {
        dispatch(startLogout());
        dispatch(noteLogout());
    }
    
    const {name} = useSelector(state => state.auth)
    const nameSort = name.split(" ")[0]

    const handleAddNew = () => {
        dispatch(startNewNote());
    }

    return (
        <div className="journal_sidebar">
            <div className = "journal_sidebar-navbar">
                <h3 className="mt-5">
                    <i className ="far fa-moon"></i>
                    <span>{nameSort}</span>
                </h3>
                <button
                className="btn"
                onClick={handleLogout}
                >Logout</button>
                
            </div>
            <div className="journal_new-entry"
                onClick={handleAddNew}
            >
                <i className="far fa-calendar-plus fa-5x"></i>
            <p
            className="mt-5"
            >New entry</p>
            </div>
            <JournalEntries />
        </div>
    )
}
