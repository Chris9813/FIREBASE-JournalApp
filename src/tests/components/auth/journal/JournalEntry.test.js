import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { activeNote } from "../../../../actions/notes";
import { JournalEntry } from "../../../../components/journal/JournalEntry";


jest.mock("../../../../actions/notes", () => ({
    activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

configure({ adapter: new Adapter() });

const initState = {};

let store = mockStore(initState);
store.dispatch = jest.fn();


const nota = {
    id: 10,
    date: 0,
    title: 'hola',
    body: 'mundo',
    url: 'https://asas.com'
};



describe('Pruebas en <JournalEntry />', () => {

    beforeEach(() => {
        store = mockStore(initState);
    });

    const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <JournalEntry {...nota}/>
        </MemoryRouter>
    </Provider>
    );
    
    test('debe e mostrarse correctamente ', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('ebe de activar la nota ', () => {
        wrapper.find('.journal_entry').prop('onClick')();
        expect(activeNote).toHaveBeenCalledWith();
    })
    
    

})
