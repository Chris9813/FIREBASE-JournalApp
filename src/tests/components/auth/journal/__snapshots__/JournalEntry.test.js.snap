// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntry /> debe e mostrarse correctamente  1`] = `
<div
  className="journal_entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal_entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://asas.com)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal_entry-body"
  >
    <p
      className="journal_entry-title"
    >
      hola
    </p>
    <p
      className="journal_entry-content"
    >
      mundo
    </p>
  </div>
  <div
    className="journal_entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
